File: App-Encapsulamento-Java/app/src/main/java/com/example/pooencapsulamento/MainActivity.java
package com.example.pooencapsulamento;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    TextView txtTituloLivro;
    TextView txtAutor;
    TextView txtEditora;
    TextView txtPaginas;
    TextView txtLocalizacao;
    Livro objLivro = new Livro();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        criarObjetos();
        iniciarComponentes();
    }

    private void iniciarComponentes() {
        txtTituloLivro = findViewById(R.id.txtTituloLivro);
        txtAutor = findViewById(R.id.txtAutor);
        txtEditora = findViewById(R.id.txtEditora);
        txtPaginas = findViewById(R.id.txtPaginas);
        txtLocalizacao = findViewById(R.id.txtLocalizacao);

        txtTituloLivro.setText(objLivro.getTitulo());
        txtAutor.setText(objLivro.getAutor());
        txtEditora.setText(objLivro.getEditora());
        txtPaginas.setText(objLivro.getPaginas());
        txtLocalizacao.setText(objLivro.getLocalizacao());

    }

    public void criarObjetos(){
            objLivro.setTitulo("Meu pé de laranja lima");
            objLivro.setAutor("José Mauro");
            objLivro.setEditora("Aliança");
            objLivro.setPaginas(121);
            objLivro.setLocalizacao("F12");
            Log.i("POO", objLivro.getTitulo());
        }
}

File: App-Encapsulamento-Kotlin/app/src/main/java/com/example/pookotlinencapsulamento/MainActivity.kt
package com.example.pookotlinencapsulamento
import android.widget.TextView;
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    lateinit var txtTitulolivro : TextView;
    lateinit var txtAutor : TextView;
    lateinit var txtEditora : TextView;
    lateinit var txtPaginas : TextView;
    lateinit var txtLocalizacao : TextView;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        iniciarComponentes();
        criarObjetos();
    }

    private fun iniciarComponentes() {
        txtTitulolivro = findViewById<TextView>(R.id.txtTituloLivro);
        txtAutor = findViewById<TextView>(R.id.txtAutor);
        txtEditora = findViewById<TextView>(R.id.txtEditora);
        txtPaginas = findViewById<TextView>(R.id.txtPaginas);
        txtLocalizacao = findViewById<TextView>(R.id.txtLocalizacao);
    }

    private fun criarObjetos() {
        var livro = Livro(
            "Kotlin",
            "Khaléu Sanches Mancini",
            "Aliança",
            "121",
            "F12");
        txtTitulolivro.setText(livro.titulo);
        txtAutor.setText(livro.autor);
        txtEditora.text = livro.editora;
        txtPaginas.text = livro.paginas;
        txtLocalizacao.text = livro.localizacao;
    }
}